dffml/service/dev.py

```diff
@@ -1056,6 +1060,96 @@ class Bump(CMD):
     packages = BumpPackages
 
 
+class SphinxBuildError(Exception):
+    """
+    Raised when sphinx-build command exits with a non-zero error code.
+    """
+
+    def __str__(self):
+        return "[ERROR] Failed run sphinx, is it installed (pip install -U .[dev])?"
+
+
+class MakeDocs(CMD):
+    """
+    Generate docs of the complete package 
+    """
+
+    async def run(self):
+
+        root = Path(__file__).parents[2]
+
+        pages_path = root / "pages"
+
+        shutil.rmtree(pages_path, ignore_errors=True)
+
+        docs_path = root / "docs"
+
+        files_to_check = [
+            ("changelog.md", "CHANGELOG.md"),
+            ("shouldi.md", "examples/shouldi/README.md"),
+            ("swportal.rst", "examples/swportal/README.rst"),
+            (
+                "contributing/consoletest.md",
+                "dffml/util/testing/consoletest/README.md",
+            ),
+        ]
+
+        for symlink, source in files_to_check:
+            file_path = docs_path / symlink
+            if not file_path.exists():
+                file_path.symlink_to(root / source)
+
+        # HTTP Service Docs
+        service_path = docs_path / "plugins/service"
+        service_path.mkdir(parents=True, exist_ok=True)
+        http_docs_pth = service_path / "http"
+        os.unlink(http_docs_pth)
+        http_docs_pth.symlink_to(root / "service/http/docs")
+
+        # Main Docs
+        runpy.run_path(root / "scripts/docs.py")
+        runpy.run_path(root / "scripts/docs_api.py")
+
+        cmd = ["sphinx-build", "-W", "-b", "html", "docs", "pages"]
```

Comment:
Let's look up the console entry point like we were doing before. This is all to avoid issues with the user's `PATH`.

Reply:
I'm not sure but I think it is fine now .... if we still need to do it some other way then please let me know :)